components/tools/OmeroJava/test/integration/chown/PermissionsTest.java

```diff
@@ -467,6 +471,571 @@ public void testChownAnnotationPrivate(boolean isDataOwner, boolean isAdmin, boo
+
+        chowner = newUserInGroup(dataGroup1, isGroupOwner);
+
+        if (isAdmin) {
+            addUsers(systemGroup, Collections.singletonList(chowner.userId), false);
+        }
+
+        /* note which objects will be used to annotate an image
+         * Note that two object sets are necessary, for the
+         * two pairs of importers, denoted as 1 and 2.*/
+        final List<IObject> annotationsAndLinksOwnAnnForTripleLinking1;
+        final List<IObject> annotationsAndLinksOwnToOthersImage1;
+        final List<IObject> annotationsAndLinksOthersToOwnImage1;
+        final List<IObject> annotationsAndLinksOthersAnnForTripleLinking1;
+        final List<IObject> annotationsOwnForTripleLinking1 = new ArrayList<IObject>();
+        final List<IObject> annotationsOthersForTripleLinking1 = new ArrayList<IObject>();
+        final List<IObject> linksOwnToOwnAnnOwnImage1 = new ArrayList<IObject>();
+        final List<IObject> linksOthersToOthersAnnOtherImage1 = new ArrayList<IObject>();
+        final List<ImageAnnotationLink> linksOwnToOthersAnnOwnImage1 = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> linksOwnToOthersAnnOthersImage1 = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> linksOthersToOwnAnnOwnImage1 = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> linksOthersToOwnAnnOthersImage1 = new ArrayList<ImageAnnotationLink>();
+        
+        /*now note the second set of the objects for image
+         * annotations
+         */
+        final List<IObject> annotationsAndLinksOwnAnnForTripleLinking2;
+        final List<IObject> annotationsAndLinksOwnToOthersImage2;
+        final List<IObject> annotationsAndLinksOthersToOwnImage2;
+        final List<IObject> annotationsAndLinksOthersAnnForTripleLinking2;
+        final List<IObject> annotationsOwnForTripleLinking2 = new ArrayList<IObject>();
+        final List<IObject> annotationsOthersForTripleLinking2 = new ArrayList<IObject>();
+        final List<IObject> linksOwnToOwnAnnOwnImage2 = new ArrayList<IObject>();
+        final List<IObject> linksOthersToOthersAnnOtherImage2 = new ArrayList<IObject>();
+        final List<ImageAnnotationLink> linksOwnToOthersAnnOwnImage2 = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> linksOwnToOthersAnnOthersImage2 = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> linksOthersToOwnAnnOwnImage2 = new ArrayList<ImageAnnotationLink>();
+        final List<ImageAnnotationLink> linksOthersToOwnAnnOthersImage2 = new ArrayList<ImageAnnotationLink>();
+
+        /* First pair of importers imports their respective images, note
+         * that importerTargetUser1 is member of 2 groups (dataGroup1 and otherGroup),
+         * so care has to be taken to import an image for this user in 
+         * each of his groups.*/
+        init(importerTargetUser1);
```

Comment:
Yes, it is the same user. This might be too clanky as you say. I tried to make sure in the first init to get the right user, in the second loginUser to get the right group (the group contex must be spelled precisely too). Probably I should have done it in some one-step method ?